### ISSUE TITLE
Investigating Apollo algorithms without a car.

### ISSUE THREAD

**Issue by pedroexenberger** [ApolloAuto/apollo#10654]
We appreciate you go through Apollo documentation and search previous issues before creating a new one. If neither of the sources helped you with your issues, please report the issue using the following form. Please note missing info can delay the response time.

### System information
**OS Platform and Distribution (e.g., Linux Ubuntu 14.04):** OS Ubuntu 16.04

**Apollo installed from (source or binary):** source

**Apollo version (1.0, 1.5, 2.0, 2.5, 3.0):** fork apollo 5.5

### **Question:**

I just arrived at this project and have some basic questions. I've looked at previous issues but didn't find anything that clearly solved my concerns.

I want to investigate the functioning of the algorithms and their required computational effort while running on CPUs and GPUs. However, I don't want to try it out in a real car. I'd just like to profile the execution of the algorithms in their respective platforms (CPU, GPU) while they receive external data from simulated sensors or pre-existent datasets.

So far I've proceeded with start guide and could execute the /apollo/docs/demo_guide/demo_3.5.record through the 'cyber_recorder play' command, while visualizing the car in the dreamview web-based platform. In this case, are the algorithms being locally executed (perception, planning, localization, etc) or the dreamview is only parsing the .record file to display the car route and decisions made during the real car trip? If the algorithms are actually performing in this case, is there a repository of more .record files?

Otherwise, what is the best way to execute the algorithms locally without having a car? I saw that there is an azure DreamLand project, but I guess this would not run locally. I also saw that there is this [LG simulator](https://www.lgsvlsimulator.com/), which apparently simulates the environment and the hardware data to feed the apollo software layer, such that I don't need a car to assert the performance of the algorithm along the simulated trip. I'm I right on any of this? Is the LG simulator + apollo the best alternative for my purposes?

Thank you very much and congrats on the project.

**Comment by ntutangyun** [comment]
Hi I'm also new here. based on my understanding:

1. To check whether the algorithms are actually executed, you may use cyber_recorder command to check whether `demo_3.5.record` contains those message channels. 
e.g. if the record file only contains sensor data, then the algorithms should be executing. 
otherwise if the record file contains those module's channels such as `/apollo/perception/obstacles`, `/apollo/localization/pose`, etc. 
Then the corresponding modules are probably not executing. 

2. I'm currently working with LGSVL and it works fine so far.

**Comment by pedroexenberger** [comment]
Hi, thank you very much for the input.

Yes, but as I could see (using cyber_record info), the demo_3.5.record file does not contain too many sensors data (such as lidar, for instance) and already have messages from several modules. Since I could not find any other .record file available, I guess the option would be to use LGSVL, or another similar simulator, to generate this data.

However, in the meantime, I found out that some modules are not open sourced yet (e.g., #4797 and #7141), which compromise the goals I'd like to achieve using the project, like instrumenting code.

Closing the issue for now.
